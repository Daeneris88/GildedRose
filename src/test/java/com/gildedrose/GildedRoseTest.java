package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class GildedRoseTest {

    @Test
    public void foo() {
        Item[] items = new Item[] { new Item("foo", 0, 0) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        assertEquals("foo", app.items[0].name);
        CombinationApprovals.verifyAllCombinations(this :: doUpdateQuality,
                new String[] {"foo", "Aged Brie", "Sulfuras, Hand of Ragnaros", "Backstage passes to a TAFKAL80ETC concert"},
                new Integer[] {0, 10, 11,6, 1},
                new Integer[] {0,  5, 2, 51, 49, 48, 50});
    }
    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}