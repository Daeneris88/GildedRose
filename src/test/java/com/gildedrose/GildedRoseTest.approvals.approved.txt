[foo, 0, 0] => foo, -1, 0 
[foo, 0, 5] => foo, -1, 3 
[foo, 0, 2] => foo, -1, 0 
[foo, 0, 51] => foo, -1, 49 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 48] => foo, -1, 46 
[foo, 0, 50] => foo, -1, 48 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 5] => foo, 9, 4 
[foo, 10, 2] => foo, 9, 1 
[foo, 10, 51] => foo, 9, 50 
[foo, 10, 49] => foo, 9, 48 
[foo, 10, 48] => foo, 9, 47 
[foo, 10, 50] => foo, 9, 49 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 5] => foo, 10, 4 
[foo, 11, 2] => foo, 10, 1 
[foo, 11, 51] => foo, 10, 50 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 48] => foo, 10, 47 
[foo, 11, 50] => foo, 10, 49 
[foo, 6, 0] => foo, 5, 0 
[foo, 6, 5] => foo, 5, 4 
[foo, 6, 2] => foo, 5, 1 
[foo, 6, 51] => foo, 5, 50 
[foo, 6, 49] => foo, 5, 48 
[foo, 6, 48] => foo, 5, 47 
[foo, 6, 50] => foo, 5, 49 
[foo, 1, 0] => foo, 0, 0 
[foo, 1, 5] => foo, 0, 4 
[foo, 1, 2] => foo, 0, 1 
[foo, 1, 51] => foo, 0, 50 
[foo, 1, 49] => foo, 0, 48 
[foo, 1, 48] => foo, 0, 47 
[foo, 1, 50] => foo, 0, 49 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 5] => Aged Brie, -1, 7 
[Aged Brie, 0, 2] => Aged Brie, -1, 4 
[Aged Brie, 0, 51] => Aged Brie, -1, 51 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 48] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 5] => Aged Brie, 9, 6 
[Aged Brie, 10, 2] => Aged Brie, 9, 3 
[Aged Brie, 10, 51] => Aged Brie, 9, 51 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 48] => Aged Brie, 9, 49 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 5] => Aged Brie, 10, 6 
[Aged Brie, 11, 2] => Aged Brie, 10, 3 
[Aged Brie, 11, 51] => Aged Brie, 10, 51 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 48] => Aged Brie, 10, 49 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 5] => Aged Brie, 5, 6 
[Aged Brie, 6, 2] => Aged Brie, 5, 3 
[Aged Brie, 6, 51] => Aged Brie, 5, 51 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 48] => Aged Brie, 5, 49 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 5] => Aged Brie, 0, 6 
[Aged Brie, 1, 2] => Aged Brie, 0, 3 
[Aged Brie, 1, 51] => Aged Brie, 0, 51 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 48] => Aged Brie, 0, 49 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5 
[Sulfuras, Hand of Ragnaros, 0, 2] => Sulfuras, Hand of Ragnaros, 0, 2 
[Sulfuras, Hand of Ragnaros, 0, 51] => Sulfuras, Hand of Ragnaros, 0, 51 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 48] => Sulfuras, Hand of Ragnaros, 0, 48 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 5] => Sulfuras, Hand of Ragnaros, 10, 5 
[Sulfuras, Hand of Ragnaros, 10, 2] => Sulfuras, Hand of Ragnaros, 10, 2 
[Sulfuras, Hand of Ragnaros, 10, 51] => Sulfuras, Hand of Ragnaros, 10, 51 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 10, 48] => Sulfuras, Hand of Ragnaros, 10, 48 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 5] => Sulfuras, Hand of Ragnaros, 11, 5 
[Sulfuras, Hand of Ragnaros, 11, 2] => Sulfuras, Hand of Ragnaros, 11, 2 
[Sulfuras, Hand of Ragnaros, 11, 51] => Sulfuras, Hand of Ragnaros, 11, 51 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 48] => Sulfuras, Hand of Ragnaros, 11, 48 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 5] => Sulfuras, Hand of Ragnaros, 6, 5 
[Sulfuras, Hand of Ragnaros, 6, 2] => Sulfuras, Hand of Ragnaros, 6, 2 
[Sulfuras, Hand of Ragnaros, 6, 51] => Sulfuras, Hand of Ragnaros, 6, 51 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49 
[Sulfuras, Hand of Ragnaros, 6, 48] => Sulfuras, Hand of Ragnaros, 6, 48 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 5] => Sulfuras, Hand of Ragnaros, 1, 5 
[Sulfuras, Hand of Ragnaros, 1, 2] => Sulfuras, Hand of Ragnaros, 1, 2 
[Sulfuras, Hand of Ragnaros, 1, 51] => Sulfuras, Hand of Ragnaros, 1, 51 
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49 
[Sulfuras, Hand of Ragnaros, 1, 48] => Sulfuras, Hand of Ragnaros, 1, 48 
[Sulfuras, Hand of Ragnaros, 1, 50] => Sulfuras, Hand of Ragnaros, 1, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 48] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 5] => Backstage passes to a TAFKAL80ETC concert, 9, 7 
[Backstage passes to a TAFKAL80ETC concert, 10, 2] => Backstage passes to a TAFKAL80ETC concert, 9, 4 
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => Backstage passes to a TAFKAL80ETC concert, 9, 51 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 48] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 5] => Backstage passes to a TAFKAL80ETC concert, 10, 6 
[Backstage passes to a TAFKAL80ETC concert, 11, 2] => Backstage passes to a TAFKAL80ETC concert, 10, 3 
[Backstage passes to a TAFKAL80ETC concert, 11, 51] => Backstage passes to a TAFKAL80ETC concert, 10, 51 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 48] => Backstage passes to a TAFKAL80ETC concert, 10, 49 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 5] => Backstage passes to a TAFKAL80ETC concert, 5, 7 
[Backstage passes to a TAFKAL80ETC concert, 6, 2] => Backstage passes to a TAFKAL80ETC concert, 5, 4 
[Backstage passes to a TAFKAL80ETC concert, 6, 51] => Backstage passes to a TAFKAL80ETC concert, 5, 51 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 48] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 5] => Backstage passes to a TAFKAL80ETC concert, 0, 8 
[Backstage passes to a TAFKAL80ETC concert, 1, 2] => Backstage passes to a TAFKAL80ETC concert, 0, 5 
[Backstage passes to a TAFKAL80ETC concert, 1, 51] => Backstage passes to a TAFKAL80ETC concert, 0, 51 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 48] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
